### DISCUSSION TITLE
custom PS1 template powerline

### DISCUSSION THREAD

**Discussion by oscarenzo** [ohmybash/oh-my-bash#342] [Q&A] [answered]
Hello,

I would like to know how can to customize the variable PS1 that by default show my prompt:

` platano  ❲p❳ ansible29  /usr/local/bin  
`

It is:

`{user}{python virtualenv}{pwd}
`

The problem in this case is when you are in directory very nested the prompt can be very large:

` earroyo  ❲p❳ ansible29  /usr/local/bin/for/bar/for/bar/path with name large/bar/for/bar/for/bar/  
`
How can I edit for the pwd not show complete, maybe just the last directory name or the two or three directories, where can I edit?, what is the var that get these values?

--
I'm using the theme 'poweline'
--

Thank you

**Comment by akinomyoga** [comment, marked as answer]
It is defined in the following place:

https://github.com/ohmybash/oh-my-bash/blob/1272a398005c9f6c43e1567078a708fd500624d2/themes/powerline/powerline.base.sh#L92-L94

Instead of directly editing the `powerline` theme file, you can define your own "segment" in `.bashrc` such as

```bash
# Your segment named "mycmd":
function __powerline_mycwd_prompt {
  echo "\w|${CWD_THEME_PROMPT_COLOR}"
}
POWERLINE_PROMPT="user_info scm python_venv ruby mycwd"
```

When you use the prompt escape sequence `\w`, you can control the number of directories shown by setting `PROMPT_DIRTRIM`. Note that this variable needs to be specified after `source "$OSH/oh-my-bash.sh"` because it overwrites the variable.

**Reply by akinomyoga** [reply]
Have you added this setting?

> ```shell
> function __powerline_mycwd_prompt {
>   echo "\w|${CWD_THEME_PROMPT_COLOR}"
> }
> POWERLINE_PROMPT="user_info scm python_venv ruby mycwd"
> ```

If it still does not work with the above setting, what would be shown when you run the following command?

```bash
$ echo "$POWERLINE_PROMPT"
```

**Comment by oscarenzo** [comment]
@akinomyoga thank you very much, I'm deleted my first comment by it was wrong, I forget the function, now it works, this is the result:

`earroyo  ❲p❳ ansible29  ../bar/for/bar/ 
`

Thank you very much again!, doubt solved :)

**NOTE:** Finally my config look like this:

```
source "$OSH"/oh-my-bash.sh

# OHMyBash custom segment it called "mycustomcwd":
function __powerline_mycustomcwd_prompt {
  echo "\w|${CWD_THEME_PROMPT_COLOR}"
}
POWERLINE_PROMPT="user_info scm python_venv ruby mycustomcwd"

PROMPT_DIRTRIM=2
```

**Reply by akinomyoga** [reply]
OK! Great!
